pkg/binding/bindingpolicy-resolution.go

```diff
@@ -215,14 +215,44 @@ func bindingObjectRefSetFromBindingWorkload(bindingSpecWorkload *v1alpha1.Downsy
 		})
 	}
 
-	return bindingObjectRefSet
+	return bindingObjectRefAndVersionSet
 }
 
-func destinationsStringSetToDestinations(destinations sets.Set[string]) []v1alpha1.Destination {
-	dests := make([]v1alpha1.Destination, 0, len(destinations))
-	for d := range destinations {
-		dests = append(dests, v1alpha1.Destination{ClusterId: d})
+func destinationsStringSetToSortedDestinations(destinationsStringSet sets.Set[string]) []v1alpha1.Destination {
+	sortedDestiantions := make([]v1alpha1.Destination, 0, len(destinationsStringSet))
+
+	// sort destinations
+	sortedDestinationStrings := destinationsStringSet.UnsortedList()
+	slices.Sort(sortedDestinationStrings)
```

Comment:
FYI:

```suggestion
	sortedDestinationStrings := destinationsStringSet.List()
```

Reply:
Please let me mark my own conversations as "resolved".

Reply:
@MikeSpreitzer apologies, it was marked resolved automatically when I committed your suggestion through the UI here.

Reply:
Oops, sorry, my suggestion was buggy. `List` is a func in the `sets` package, not a method of `sets.Set`.

Reply:
Was wondering how I missed that method. So reverting is sufficient?

Reply:
Reverting would work. So would changing the code to the following.

```go
	sortedDestinationStrings := sets.List(destinationsStringSet)
```

Reply:
Done.